commit 3dc1387c53210c18c904b6e95ccc3dcf56a9eacd (refs/changes/34/2534/2)
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2017-09-09T14:24:13-07:00 (2 years, 1 month ago)
    
    IMGAPI-642 missing file0 from imgapi docker image causing container provision failure
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/CHANGES.md b/CHANGES.md
index 06a0bdc..e51c40d 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -1,5 +1,10 @@
 # IMGAPI changelog
 
+## 4.0.7
+
+- IMGAPI-642 missing file0 from imgapi docker image causing container provision
+  failure
+
 ## 4.0.6
 
 - IMGAPI-643 docker pull crashes if an unexpected x-registry-config header is
diff --git a/lib/docker.js b/lib/docker.js
index 9a58578..6aa866b 100644
--- a/lib/docker.js
+++ b/lib/docker.js
@@ -1220,34 +1220,6 @@ function _dockerV22Pull(ctx, cb) {
             next();
         },
 
-        function addSdcDockerImage(_, next) {
-            // Create the sdc-docker image in the docker_images bucket.
-
-            // Calculate total size and find the last image uuid.
-            var finalUuid;
-            var size = ctx.layerInfos.map(function (layer) {
-                if (!layer.imgManifest) {
-                    return 0;
-                }
-                assert.object(layer.imgFile, 'layer.imgFile');
-                finalUuid = layer.imgManifest.uuid;
-                return layer.imgFile && layer.imgFile.size || 0;
-            }).reduce(function (a, b) { return a + b; }, 0);
-
-            ctx.resMessage({
-                type: 'create-docker-image',
-                config_digest: configDigest,
-                head: true,
-                image: imgJson,
-                image_uuid: finalUuid,
-                manifest_digest: ctx.manifestDigest,
-                manifest_str: ctx.manifestStr,
-                size: size,
-                dockerImageVersion: ctx.dockerImageVersion
-            });
-            next();
-        },
-
         /*
          * *Serially* complete the import of all the images:
          * - We only ActivateImage's at this stage after the file downloading
@@ -1275,6 +1247,34 @@ function _dockerV22Pull(ctx, cb) {
             });
         },
 
+        function createSdcDockerImage(_, next) {
+            // Create the sdc-docker image in the docker_images bucket.
+
+            // Calculate total size and find the last image uuid.
+            var finalUuid;
+            var size = ctx.layerInfos.map(function (layer) {
+                if (!layer.imgManifest) {
+                    return 0;
+                }
+                assert.object(layer.imgFile, 'layer.imgFile');
+                finalUuid = layer.imgManifest.uuid;
+                return layer.imgFile && layer.imgFile.size || 0;
+            }).reduce(function (a, b) { return a + b; }, 0);
+
+            ctx.resMessage({
+                type: 'create-docker-image',
+                config_digest: configDigest,
+                head: true,
+                image: imgJson,
+                image_uuid: finalUuid,
+                manifest_digest: ctx.manifestDigest,
+                manifest_str: ctx.manifestStr,
+                size: size,
+                dockerImageVersion: ctx.dockerImageVersion
+            });
+            next();
+        },
+
         function finishingMessages(_, next) {
             resMessage({
                 type: 'status',
diff --git a/package.json b/package.json
index 13e784e..f8bf4ae 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "imgapi",
   "description": "Image API to manage images for SDC 7",
-  "version": "4.0.6",
+  "version": "4.0.7",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
